judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/submission/programming/ContestSubmissionResource.java

```diff
@@ -228,6 +234,29 @@ public SubmissionWithSourceResponse getSubmissionWithSourceById(
                 .build();
     }
 
+    @Override
+    @UnitOfWork(readOnly = true)
+    public Response getSubmissionSourceImage(String contestJid, String userJid, String problemJid) {
+        Contest contest = checkFound(contestStore.getContestByJid(contestJid));
+        checkAllowed(scoreboardRoleChecker.canViewOtherContestantSolution(contest));
+
+        List<Submission> submissions = submissionStore
```

Comment:
This is not efficient. Can we create something like `SubmissionStore::getLatestSubmission(userJid, problemJid)`? It can reuse `BaseProgrammingSubmissionDao::selectPaged()` with page size = 1.